Switching the Empathy Workshop to use the VCS workflow for the stark-enterprises workspace

shernandez5 opened the pull request:
I've updated the workflow to use the VCS workflow for the stark-enterprises repo, which means the users will need to configure the VCS workflow, but that is a little more of a guided process than requiring the user to generate a PAT.

Testers of our initial workflow mentioned the process to install Terraform on their local CLIs was tedious, and there were a lot of steps to get set up to use the CLI workflow. So I updated the steps to instead use the VCS workflow, which reduced the number of steps significantly. So now users must configure their VCS connection with a specific name, and we will use a data source to look up that oauth client and grab its oauth_token_id

shernandez5 committed:
Testing without the cloud block and using VCS instead

shernandez5 committed:
remove creation of vcs connection

sebasslash commented:
Could I get some insight on the removal of the oauth client (vcs) module? There seems to be some confusion on what `oauth_token_id` is, this is a resource in TFC not a GitHub PAT. Unless the intent here is to manually create these OAuth clients and provide each group their corresponding token ID. But that process seems more tedious than allowing each "team lead" (the person who forks these repos) to create a PAT. This also means people would work off branches, since these oauth tokens would be tied to the same Github account. 

I would be more in favor having teams work on their forks rather than creating/managing branches on the same repo. It's less manual configuration on our part, less error prone on theirs.

shernandez5 commented:
Yeah, sorry I should have given a bit more detail. I've updated the workflow to use the VCS workflow for the stark-enterprises repo, which means the users will need to configure the VCS workflow, but that is a little more of a guided process than requiring the user to generate a PAT.

Bruce and Marisa also mentioned that the process to install Terraform on their local CLIs was tedious, and there were a lot of steps to get set up to use the CLI workflow. So I updated the steps to instead use the VCS workflow, which reduced the number of steps significantly. So now once the user configures their VCS connection, they just need to copy the OAuth token and pass that into their workspace.

shernandez5 committed:
grab vcs oauth token via data source

shernandez5 committed:
Merge pull request #1 from shernandez5/vcs-testing

grab vcs oauth token via data source
